torchmetrics/functional/image/ms_ssim.py

```diff
@@ -0,0 +1,204 @@
+        sigma: Standard deviation of the gaussian kernel (default: (1.5, 1.5))
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
+
+        data_range: Range of the image. If ``None``, it is determined from the image (max - min)
+        k1: Parameter of SSIM.
+        k2: Parameter of SSIM.
+        betas: Exponent parameters for individual similarities and contrastive sensitivies returned by different image
+            resolutions.
+        normalize: When MultiScaleSSIM loss is used for training, it is desirable to use normalizes to improve the
+            training stability. This `normalize` argument is out of scope of the original implementation [1], and it is
+            adapted from https://github.com/jorge-pessoa/pytorch-msssim instead.
+
+    Raises:
+        ValueError:
+            If the image height or width is smaller then ``2 ** len(betas)``.
+        ValueError:
+            If the image height is smaller than ``(kernel_size[0] - 1) * max(1, (len(betas) - 1)) ** 2``.
+        ValueError:
+            If the image width is smaller than ``(kernel_size[0] - 1) * max(1, (len(betas) - 1)) ** 2``.
+    """
+    sim_list: List[Tensor] = []
+    cs_list: List[Tensor] = []
+
+    if preds.size()[-1] < 2 ** len(betas) or preds.size()[-2] < 2 ** len(betas):
+        raise ValueError(
+            f"For a given number of `betas` parameters {len(betas)}, the image height and width dimensions must be "
+            f"larger than or equal to {2 ** len(betas)}."
+        )
+
+    _betas_div = max(1, (len(betas) - 1)) ** 2
+    if preds.size()[-2] // _betas_div <= kernel_size[0] - 1:
+        raise ValueError(
+            f"For a given number of `betas` parameters {len(betas)} and kernel size {kernel_size[0]}, the image height "
+            f"must be larger than {(kernel_size[0] - 1) * _betas_div}."
+        )
+    if preds.size()[-1] // _betas_div <= kernel_size[1] - 1:
+        raise ValueError(
+            f"For a given number of `betas` parameters {len(betas)} and kernel size {kernel_size[1]}, the image width "
+            f"must be larger than {(kernel_size[1] - 1) * _betas_div}."
+        )
+
+    for _ in range(len(betas)):
+        sim, contrast_sensitivity = _get_normalized_sim_and_cs(
+            preds, target, kernel_size, sigma, reduction, data_range, k1, k2, normalize
+        )
+        sim_list.append(sim)
+        cs_list.append(contrast_sensitivity)
+        preds = F.avg_pool2d(preds, (2, 2))
+        target = F.avg_pool2d(target, (2, 2))
+
+    sim_stack = torch.stack(sim_list)
+    cs_stack = torch.stack(cs_list)
+
+    if normalize == "simple":
+        sim_stack = (sim_stack + 1) / 2
+        cs_stack = (cs_stack + 1) / 2
+
+    sim_stack = sim_stack ** torch.tensor(betas)
+    cs_stack = cs_stack ** torch.tensor(betas)
```

Comment:
locally got some device errors here
```suggestion
    sim_stack = sim_stack ** torch.tensor(betas, device=sim_stack.device)
    cs_stack = cs_stack ** torch.tensor(betas, device=cs_stack.device)
```